src/main.rs

```diff
@@ -120,17 +140,29 @@ fn run_multi_threaded(parameters_vec: Vec<Parameters>, output_path: &Path, threa
             >(
                 sender, Scenario { scenario }
             ));
+            context.set_report_item_handler::<DeathReport>(get_channel_report_handler::<
+                DeathReport,
+                Scenario,
+            >(
+                death_sender, Scenario { scenario }
+            ));
             setup_context(&mut context, &parameters);
             context.execute();
             println!("Scenario {} completed", scenario);
         });
     }
     drop(sender);
+    drop(death_sender);
 
     // Write output from main thread
-    let mut writer = csv::Writer::from_writer(output_file);
+    let mut incidence_writer = csv::Writer::from_writer(output_file);
```

Comment:
As @jasonasher observed privately, this actually buffers all the death reports till all the incidence reports are done. Is there some way to listen to the next one on either channel?

Reply:
Maybe using tokio select!

Reply:
Trying this now